tests/suites/storage/named_storage_pool.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# Test that storage from a named storage pool can be used when deploying an application.
+# This test verifies that when a storage pool is created with create-storage-pool,
+# it can be referenced in the storage directive when deploying applications.
+run_named_storage_pool() {
+	echo
+
+	model_name="named-storage-pool"
+	file="${TEST_DIR}/test-${model_name}.log"
+
+	ensure "${model_name}" "${file}"
+
+	# Create a named storage pool
+	juju create-storage-pool -m "${model_name}" roottest rootfs size=100M
+	juju create-storage-pool -m "${model_name}" temptest tmpfs size=100M
+
+	# Verify the storage pools are created
+	juju list-storage-pools -m "${model_name}" --format json | jq '.roottest | .provider' | check "rootfs"
+	juju list-storage-pools -m "${model_name}" --format json | jq '.temptest | .provider' | check "tmpfs"
+
+	# Deploy the application with the storage pool for rootfs
+	juju deploy -m "${model_name}" ./testcharms/charms/dummy-storage --storage single-fs=roottest,1,10M
+
+	# Wait for the application to be active.
+	wait_for "dummy-storage" "$(active_condition "dummy-storage" 0)"
+	wait_for "active" "$(workload_status "dummy-storage" 0).current"
+
+	# Verify tmpfs storage details.
+	wait_for_storage "attached" '.storage["single-fs/0"]["status"].current'
+	assert_storage "filesystem" "$(kind_name "single-fs" 0)"
+	assert_storage "alive" "$(life_status "single-fs" 0)"
+	assert_storage "dummy-storage/0" "$(unit_attachment "single-fs" 0 0)"
+
+	# Verify the application use the correct storage pool
+	assert_storage "roottest" '.filesystems["0/0"]["pool"]'
+
+	# Clean up.
+	juju remove-application -m "${model_name}" --force --no-prompt dummy-storage
+	wait_for "{}" ".applications"
+
+	# Deploy the application with the storage pool for tmpfs
+	juju deploy -m "${model_name}" ./testcharms/charms/dummy-storage --storage single-fs=temptest,1,10M
+
+	# Wait for the application to be active.
+	wait_for "dummy-storage" "$(active_condition "dummy-storage" 0)"
+	wait_for "active" "$(workload_status "dummy-storage" 1).current"
+
+	# Verify tmpfs storage details.
+	wait_for_storage "attached" '.storage["single-fs/1"]["status"].current'
+	assert_storage "filesystem" "$(kind_name "single-fs" 1)"
+	assert_storage "alive" "$(life_status "single-fs" 1)"
+	assert_storage "dummy-storage/1" "$(unit_attachment "single-fs" 1 0)"
+
+	# Verify the application use the correct storage pool
+	assert_storage "temptest" '.filesystems["1/1"]["pool"]'
+
+	# Clean up.
+	juju remove-application -m "${model_name}" --force --no-prompt dummy-storage
```

Comment:
No need to remove the app - the model is being destroyed

Reply:
Got it. Let me extend the existing tests instead creating new.